### PR TITLE
London | Saba Farjamfard |Module-Data-Groups | Sprint 1 | Week 1

### PR DESCRIPTION
**Pull request by sabafarjamfard**
## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/contributing/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: Sprint-1/implement/dedupe.test.js [hunk 1]
```diff
@@ -3,25 +3,43 @@ const dedupe = require("./dedupe.js");
...
 test.todo("given an empty array, it returns an empty array");
-
 // Given an array with no duplicates
 // When passed to the dedupe function
 // Then it should return a copy of the original array
-
 // Given an array with strings or numbers
 // When passed to the dedupe function
 // Then it should remove the duplicate values, preserving the first occurence of each element
+const dedupe = require('./dedupe'); //
```
Comment (cjyuan) [inline Sprint-1/implement/dedupe.test.js:23]: Have you tried executing this test script? I see some typo in the code which I think should prevent this script from executing properly.

File: Sprint-1/implement/sum.test.js [hunk 2]
```diff
@@ -34,3 +29,31 @@ test.todo("given an empty array, returns 0")
...
+  });
+
+  test('ignores non-number values and returns the sum of numerical elements', () => {
+    expect(sum(['a', 10, 'b', 20, null, 30])).toBe(60);
+  });
+
+  test('returns 0 for an array with only non-number values', () => {
+    expect(sum(['a', 'b', {}, [], null])).toBe(0);
+  });
+});
```
Comment (cjyuan) [inline Sprint-1/implement/sum.test.js:59]: What do you expect from the following function calls (on extreme cases)?
Does your function returns the value you expected?

```
sum([Infinity, 1]);
sum([Infinity, Infinity]);
sum([NaN, 1]);
sum([Infinity, -Infinity]);
```

**Review by cjyuan — COMMENTED** [review]
Code looks great. But it seems like you have a typo in your code that would prevent it from executing properly.

File: Sprint-1/implement/dedupe.test.js (hunk 1 above)
Comment (sabafarjamfard) [reply Sprint-1/implement/dedupe.test.js:23]: hi 
Thanks for pointing out the issue,  I have executed the test, and all the tests have passed without errors.

File: Sprint-1/implement/sum.test.js (hunk 2 above)
Comment (sabafarjamfard) [reply Sprint-1/implement/sum.test.js:75]: Hi,
Thanks for pointing out those edge cases. I’ve reviewed and updated my sum function to handle the cases you mentioned:
	1.	sum([Infinity, 1]) → Now correctly returns Infinity.
	2.	sum([Infinity, Infinity]) → Now correctly returns Infinity.
	3.	sum([NaN, 1]) → Now correctly returns NaN.
	4.	sum([Infinity, -Infinity]) → Now correctly returns NaN.

I’ve also added corresponding tests to ensure these scenarios are covered and verified that all tests pass successfully.
